test/typesArray.test.js

```diff
@@ -417,3 +417,22 @@ test('object that is simultaneously a string and a json switched', (t) => {
   const valueObj = stringify({ simultaneously: { foo: likeObjectId } })
   t.is(valueObj, '{"simultaneously":{"foo":"hello"}}')
 })
+
+test('should default to [] when type is array and object is null', (t) => {
+  t.plan(1)
+  const schema = {
+    type: 'object',
+    properties: {
+      arr: {
+        type: 'array',
+        items: {
+          type: 'number'
+        }
+      }
+    }
+  }
```

Comment:
Are you sure passing null for the array would validate this schema? Can you check with ajv?

My understanding is that null is not a valid array.

Reply:
```javascript
const schema = {
  type: 'object',
  properties: {
    foo: {
      $id: '#/properties/foo',
      type: 'array',
      default: [],
      items: {
        $id: '#/properties/foo/items'
      }
    }
  }
}
const obj = {
  foo: null
}
var Ajv = require('ajv')
var ajv = new Ajv()
var validate = ajv.compile(schema)
var valid = validate(obj)
if (!valid) console.log(validate.errors)
// output
// [
//   {
//     keyword: 'type',
//     dataPath: '.foo',
//     schemaPath: '#/properties/foo/type',
//     params: { type: 'array' },
//     message: 'should be array'
//   }
// ]

```
Indeed